bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/Auditor.java

```diff
@@ -196,20 +204,39 @@ public void run() {
 
                         if (lostBookies.size() > 0) {
                             knownBookies.removeAll(lostBookies);
-
-                            auditBookies();
+                            if (conf.getLostBookieRecoveryDelay() == 0 ||
+                                lostBookies.size() > 1 ||
+                                auditTask != null) {
+                                // 1) If more than one bookie is down, start the audit immediately;
+                                // 2) If we had scheduled an audit earlier for a lost bookie and in
+                                //    the meantime another bookie goes down, let us not delay recovery;
+                                //    we cancel the previously scheduled audit before starting the new one
+                                if (auditTask != null) {
+                                    if (auditTask.cancel(false)) {
```

Comment:
The documentation for cancel return code says  "[it returns] false if the task could not be cancelled, typically because it has already completed normally; true otherwise"

So I decided to schedule another one just to be safe. Running it twice shouldn't cause any issues though. Let me know if you want me to not schedule one if cancel fails.